internal/controllers/machine_controller.go

```diff
@@ -131,6 +135,7 @@ type MachineReconciler struct {
 
 	machines      store.Store[*api.Machine]
 	machineEvents event.Source[*api.Machine]
+	eventRecorder machineEvent.EventRecorder
```

Comment:
Maybe just embed the EventRecorder in order to let us call it directly like `r.Eventf(....) `
```suggestion
	machineEvent.EventRecorder
```

Reply:
Done